examples/run_config.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+"""
+Using :class:`~caf.toolkit.BaseConfig`
+======================================
+
+The :class:`~caf.toolkit.BaseConfig` class in caf.toolkit is designed to load
+and validated YAML configuration files. This example shows how
+to create child classes of BaseConfig to load parameters.
+
+See Also
+--------
+:class:`~caf.toolkit.arguments.ModelArguments`
+    for producing command-line arguments from a BaseConfig class.
+"""
+
+# Built-Ins
+import pathlib
+import re
+from typing import Any, Self
+
+# Third Party
+import pydantic
+from pydantic import dataclasses
+
+# Local Imports
+import caf.toolkit as ctk
+
+folder = pathlib.Path().parent
+
+# %%
+# Basic
+# -----
+#
+# This example shows how to create a simple configuration file, without nesting, and
+# with different types of parameters. The majority of Python built-in types can be used,
+# additionally dataclasses and many "simple"[#simple]_ custom types can be used.
+#
+# .. seealso ::
+#       :ref:`Extra Validation` for information on more complex validation.
+#
+# The :class:`pydantic.DirectoryPath` and :class:`pydantic.FilePath` types both return
+# :class:`pathlib.Path` objects after validating that the directory, or file, exists.
+
+
+class Config(ctk.BaseConfig):
+    """Example of a basic configuration file without nesting."""
+
+    years: list[int]
+    name: str
+    output_folder: pydantic.DirectoryPath
+    input_file: pydantic.FilePath
+
+
+# %%
+# Example of the YAML config file which is loaded by the above class.
+#
+# .. literalinclude:: /../../examples/basic_config.yml
+#       :language: yaml
+#       :caption: Config file: examples/basic_config.yml
+#
+# Below shows how to load the config file and displays the class as text.
+
+parameters = Config.load_yaml(folder / "basic_config.yml")
+print(parameters)
+
```

Comment:
I think it would be good at this point to include the "Note" box, which basically notes a few interesting things about the example above, e.g.,
* How the `DirectoryPath` and `FilePath` types work
* That the names in the Config class need to be the same in the YAML, but the order can be different.

I think it's better after the example so that reader doesn't get overwhelmed with too much information when reading this example.